src/include/rbd/librbd.h

```diff
@@ -1179,6 +1179,7 @@ CEPH_RBD_API int rbd_aio_flush(rbd_image_t image, rbd_completion_t c);
  * @returns 0 on success, negative error code on failure
  */
 CEPH_RBD_API int rbd_invalidate_cache(rbd_image_t image);
+CEPH_RBD_API int rbd_invalidate_cache2(rbd_image_t image, bool discard);
```

Comment:
What's the purpose of the `discard` parameter? The existing `invalidate_cache` method will drop all the data in the cache (i.e. lose any uncommitted writeback).